### PR TITLE
ISPN-11913 search and improved index

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-11913

### CODE REVIEW

File: documentation/src/main/asciidoc/index.asciidoc [hunk 1]
```diff
@@ -2,60 +2,58 @@
 Written and maintained by the {brandname} community
 :icons: font
 
-Get answers and find help in the {brandname} {infinispanversion} doc set.
+++++
+<script async src="https://cse.google.com/cse.js?cx=017043654585294774899:cdrjpvj5awo"></script>
+<div class="gcse-search"></div>
+++++
```
Comment (oraNod) [inline documentation/src/main/asciidoc/index.asciidoc:8]: @tristantarrant is this search option cool? here's the public url: https://cse.google.com/cse?cx=017043654585294774899:cdrjpvj5awo 

it includes https://infinispan.org/docs/stable/* and /docs/dev/* in the search result. it works fairly well but I don't know if this is the best solution. another thing is that I created this with my personal gmail account, which might not be ideal.

File: documentation/src/main/asciidoc/index.asciidoc [hunk 2]
```diff
@@ -2,60 +2,58 @@
...
+[discrete]
+== Developer
 
-* link:titles/hotrod_java/hotrod_java.html[Hot Rod Java Client Guide]
+[unstyled]
+* link:titles/embedding/embedding.html[Embedding {brandname}]
+* link:titles/developing/developing.html[Developing for {brandname}]
+* link:titles/configuring/configuring.html[Configuring {brandname}]
+* link:titles/xsite/xsite.html[Using Cross-Site Replication]
+* link:titles/hotrod_java/hotrod_java.html[Hot Rod Java Clients]
```
Comment (gustavonalle) [inline documentation/src/main/asciidoc/index.asciidoc:35]: Can you add querying in the front page?

File: documentation/src/main/asciidoc/index.asciidoc (hunk 1 above)
Comment (gustavonalle) [reply documentation/src/main/asciidoc/index.asciidoc:8]: I am not sure what is wrong with this particular search box, but it doesn't seem to find stuff :)
For example, I tried searching for "auto-config", "InfinispanIndexManager", "Query" and did not get any interesting result.
OTOH, the one from https://cse.google.com/cse?cx=013887093884325579276:mkumderkeqo do bring me all those results

Comment (gustavonalle) [reply documentation/src/main/asciidoc/index.asciidoc:8]: "local-heap" is another term it does not find meaningful results. Also, it appears the operator guide is not being indexed by the search

File: documentation/src/main/asciidoc/index.asciidoc (hunk 2 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/index.asciidoc:35]: @gustavonalle Based on feedback from Tristan, I moved querying from it's own title into the Developer's Guide. We could add a link to the anchor like this:

link:titles/developing/developing.html#indexing_querying[Indexing and Querying]

File: documentation/src/main/asciidoc/index.asciidoc (hunk 1 above)
Comment (tristantarrant) [reply documentation/src/main/asciidoc/index.asciidoc:8]: we should probably use an infinispan.org account for this

**Pull request merged**